### PR TITLE
Mf/add dao coin limit order postgres skeleton

### PR DESCRIPTION
**Pull request by mattfoley8**

### CODE REVIEW

**Comment by mattfoley8** [comment]
Have to check out failing CI test.

**Comment by lazynina** [comment]
> I have to check out the failing CI test.

It appears to have been a transient issue - I just re-ran the tests and it passed.

File: lib/postgres.go [hunk 1]
```diff
@@ -391,6 +392,19 @@ type PGMetadataDerivedKey struct {
...
+type PGMetadataDAOCoinLimitOrder struct {
+	tableName struct{} `pg:"pg_metadata_dao_coin_limit_order"`
+
+	TransactionHash            *BlockHash `pg:",pk,type:bytea"`
+	DenominatedCoinType        uint32     `pg:",use_zero"`
+	DenominatedCoinCreatorPKID *PKID      `pg:",type:bytea"`
+	DAOCoinCreatorPKID         *PKID      `pg:",type:bytea"`
+	OperationType              uint32     `pg:",use_zero"`
+	PriceNanos                 uint64     `pg:",use_zero"`
+	Quantity                   uint64     `pg:",use_zero"`
```
Comment (lazynina) [inline lib/postgres.go:405]: Use a string to represent uint256's. See `PGMetadataDAOCoinTransfer` and how it uses `FromHex` to convert a hex string to a Uin2565 when reading from the database. and uses `uint256.Hex()` when creating a `PGMetadataDAOCoinTransfer` struct

File: lib/postgres.go [hunk 2]
```diff
@@ -1173,6 +1188,12 @@ func (postgres *Postgres) InsertTransactionsTx(tx *pg.Tx, desoTxns []*MsgDeSoTxn
 				ReceiverPublicKey:      txMeta.ReceiverPublicKey,
 			})
 
+		} else if txn.TxnMeta.GetTxnType() == TxnTypeDAOCoinLimitOrder {
+			txMeta := txn.TxnMeta.(*DAOCoinLimitOrderMetadata)
+			_ = txMeta
+			_ = metadataDAOCoinLimitOrder
```
Comment (lazynina) [inline lib/postgres.go:1194]:
```suggestion
			_, _ = txMeta, metadataDAOCoinLimitOrder
```
you can keep the changes to a single line this way

File: migrate/20220309224100_create_dao_coin_limit_order_tables.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	up := func(db orm.DB) error {
+		_, err := db.Exec(`
+			CREATE TABLE pg_metadata_dao_coin_limit_orders (
+				transaction_hash              BYTEA PRIMARY KEY,
+				denominated_coin_type         SMALLINT NOT NULL,
+				denominated_coin_creator_pkid BYTEA NOT NULL,
+				dao_coin_creator_pkid         BYTEA NOT NULL,
+				operation_type                SMALLINT NOT NULL,
+				price_nanos                   BIGINT NOT NULL,
+				quantity                      BIGINT NOT NULL
```
Comment (lazynina) [inline migrate/20220309224100_create_dao_coin_limit_order_tables.go:18]:
```suggestion
				price_nanos                   TEXT NOT NULL,
				quantity                      TEXT NOT NULL
```

File: lib/postgres.go [hunk 4]
```diff
@@ -391,6 +392,19 @@ type PGMetadataDerivedKey struct {
...
 }
 
+// PGMetadataDAOCoinLimitOrder represents DAOCoinLimitOrderMetadata
+type PGMetadataDAOCoinLimitOrder struct {
+	tableName struct{} `pg:"pg_metadata_dao_coin_limit_order"`
+
+	TransactionHash            *BlockHash `pg:",pk,type:bytea"`
+	DenominatedCoinType        uint32     `pg:",use_zero"`
+	DenominatedCoinCreatorPKID *PKID      `pg:",type:bytea"`
+	DAOCoinCreatorPKID         *PKID      `pg:",type:bytea"`
```
Comment (lazynina) [inline lib/postgres.go:402]:
```suggestion
	DAOCoinCreatorPKID         *PKID      `pg:"dao_coin_creator_pkid,type:bytea"`
```
Postgres infers the column names based on the capitalization of the field name. You can override the default behavior by specifying the name in the struct tag. Having PKID at the end is fine for whatever reason and is parsed as `_pkid` so no need to change those fields.

**Review by lazynina — CHANGES_REQUESTED** [review]

**Review by lazynina — APPROVED** [review]

**Pull request merged**